packages/geoview-core/src/core/components/app-bar/buttons/geolocator.tsx

```diff
@@ -0,0 +1,49 @@
+import { useState } from 'react';
+
+import { api, booleanPayload } from '../../../../app';
+import { EVENT_NAMES } from '../../../../api/events/event-types';
+import { IconButton, SearchIcon } from '../../../../ui';
+
+/**
+ * Interface used for geolocator button properties
+ */
+interface GeolocatorProps {
+  className?: string | undefined;
```

Comment:
I recommend to have the mandatory props first, then optional props after that. For example in this case you can have mapId first, then after that className as it's optional.